crates/bcr-wallet-core/src/wallet.rs

```diff
@@ -160,26 +152,58 @@ where
         let credit = self.credit.balance().await?;
         Ok(WalletBalance { debit, credit })
     }
-    async fn prepare_send_with_pocket(
-        amount: Amount,
-        keysets_info: &[KeySetInfo],
-        pocket: &dyn Pocket,
-    ) -> Result<(SendReference, SendSummary)> {
-        let pocket_summary = pocket.prepare_send(amount, keysets_info).await?;
-        let reference = SendReference {
-            request_id: Uuid::new_v4(),
-            amount,
-            unit: pocket.unit(),
-            internal_rid: pocket_summary.request_id,
+
+    async fn check_nut18_request(
+        &self,
+        req: &cdk18::PaymentRequest,
+    ) -> Result<(Amount, CurrencyUnit, cdk18::Transport)> {
+        if let Some(mints) = &req.mints {
+            if !mints.contains(&self.mint_url) {
+                return Err(Error::InterMint);
+            }
+        }
+        if req.nut10.is_some() {
+            return Err(Error::SpendingConditions);
+        }
+        let Some(amount) = req.amount else {
+            return Err(Error::MissingAmount);
+        };
+        let unit = if let Some(unit) = &req.unit {
+            if *unit != self.debit.unit() && *unit != self.credit.unit() {
+                return Err(Error::CurrencyUnitMismatch(self.debit.unit(), unit.clone()));
+            }
+            unit.clone()
+        } else if amount <= self.credit.balance().await? {
+            self.credit.unit()
+        } else {
+            self.debit.unit()
         };
-        let summary = SendSummary {
-            request_id: reference.request_id,
-            amount,
-            unit: pocket.unit(),
-            send_fees: pocket_summary.send_fees,
-            swap_fees: pocket_summary.swap_fees,
+        let empty = vec![];
+        let transports = req.transports.as_ref().unwrap_or(&empty);
+        let (nostrs, https): (Vec<_>, Vec<_>) = transports
```

Comment:
isn't it `nostr` and `https`? or is the `s` for plurality of httpS and nostrS?
I guess it's for plurality, but I was a bit confused because `https` has a distinct meaning 😅 maybe `nostr_tsports`/`http_tsports` or smth?

Not too important, though